### ISSUE TITLE
Book about Data Compression?

### ISSUE THREAD

**Issue by schierlm** [github/archive-program#86]
I don't know how comprehensive the documentation of the XZ format included on every reel is, but if it is just the C source code of `xzdec` or a similar tool, it would be prudent to include some book about data compression concepts (huffman coding, arithmetic coding, LZ77, etc.). While it should be possible to port any program from any language to any other language without understanding *why* it works, this is certainly easier when you have some background what the program is doing.

**Comment by nferraz** [comment]
Huffman coding is covered by data communication and networking.

Data compression is well covered in wikipedia, including, for example, [HTTP data compression](https://en.wikipedia.org/wiki/HTTP_compression).

Specific algorithms, such as GZIP, are covered by RFCs (e.g. [RFC 1952](https://tools.ietf.org/html/rfc1952)).

**Comment by schierlm** [comment]
> [Huffman coding](https://en.wikipedia.org/wiki/Huffman_coding), [Arithmetic coding](https://en.wikipedia.org/wiki/Arithmetic_coding) and [LZ77/LZ78](https://en.wikipedia.org/wiki/LZ77_and_LZ78) are well covered in wikipedia.

I agree, the [Data compression](https://en.wikipedia.org/wiki/Category:Data_compression) category in Wikipedia is extensive enough.

However, as I understand it, the Wikipedia dump will be included as XZ-encoded "QR codes" on the reels, like the rest of the repository data. Therefore, citing it for descriptions how to decode itself will cause kind of a chicken and egg problem.

And this also ignores the fact that you need some more computer science knowledge (like how to build a B-tree index) to be even able to efficiently work with dozens-of-gigabytes large Wikipedia dumps containing millions of articles.

In other words, while I agree that Wikipedia should be included (it may be debated whether the raw enwiki dump or some preprocessed form that already substituted the templates, like the ones from aarddict.org, would be the better choice), I do not think you can use it as an argument to reduce the "fundamental information" included in unencoded form in the Tech Tree.

**Comment by edsu** [comment]
This might be a good argument for some other more accessible and simplified representation of Wikipedia being put into the Tech Tree? I imagine it would be pretty useful in other areas as people tried to piece things back together.

**Comment by rezendi** [comment]
This is a good point - thanks.
